### PR TITLE
Fix and improve `Geocaching.search_rect()`

### PR DESCRIPTION
**Pull request by BelKed**
Since we can pass parameters for reversing and limiting the returned caches directly to the URL requested by `Geocaching.search_rect()`, I believe it would be helpful (at least for me) to have these parameters accessible in Python code.

I also fixed the parsing of the `userFound` property because the returned data format has changed a bit.

And, of course, I reloaded the test cartridges to reflect recent changes.

### CODE REVIEW

**Comment by BelKed** [comment]
> Could you please add a test for the reverse sort order as well?

Sure, I added it right now in https://github.com/tomasbedrich/pycaching/pull/203/commits/14db0c7a4679d899d119b717d5aef4314de2ec0d. Is it fine?

**Review by FriedrichFroebel — CHANGES_REQUESTED** [review]

File: test/test_geocaching.py [hunk 1]
```diff
@@ -142,6 +142,22 @@ def test_search_rect(self):
...
+                    waypoints = {cache.wp for cache in caches}
+                    self.assertSetEqual(waypoints, expected)
+
+                    # Check if caches are reverse sorted by distance to origin
+                    distances = []
+                    for cache in caches:
+                        try:
+                            distances.append(great_circle(cache.location, origin).meters)
+                        except PMOnlyException:
+                            # can happend when getting accurate location
```
[the remark this answers is not in the document]
Comment (BelKed) [reply test/test_geocaching.py]: Oh, didn't realise that...
Fixed in https://github.com/tomasbedrich/pycaching/pull/203/commits/3d98e6c7482e9ca79d0364919c94a326711a4a22.

[the remark this answers is not in the document]
Comment (BelKed) [reply test/test_geocaching.py]: Oh, didn't realise that...
Fixed in https://github.com/tomasbedrich/pycaching/pull/203/commits/3d98e6c7482e9ca79d0364919c94a326711a4a22.

**Review by FriedrichFroebel — APPROVED** [review]
Thanks for your patience.

**Pull request merged**